frontend/src/stores/levels/actions.ts

```diff
@@ -0,0 +1,17 @@
+import { levels } from './state'
+import { get } from '@/services/http'
+
+export async function fetchLevelsByRoleId(roleId: string) {
+  const data: any[] | string = await get(`/levels?role_id=${roleId}`)
+  if (Array.isArray(data)) {
+    levels.value = data.map((level: any) => ({
```

Comment:
Please, put these levels in the role state. Each role has its levels, so you can update there, and get rid of levels state file